pkg/config/config.go

```diff
@@ -16,13 +16,13 @@ func LoadFile(filename string, v any) error {
 		return err
 	}
 
-	dec, ok := NewDecoderByFilename(filename)
-	if !ok {
+	dec, _ := NewDecoderByFilename(filename)
+	if dec == nil {
 		// fallback to autodetect
-		dec, ok = NewDecoder("auto")
+		dec, _ = NewDecoder("auto")
 	}
 
-	if !ok {
+	if dec == nil {
 		return ErrUnknownFormat
```

Comment:
should it be wrapped including the filename?

Reply:
it is unreachable because `NewDecoder("auto")` never fails, but it's not a strong _never_. `Autodetect` might not be registered.